app/javascript/components/physical-storage-form/index.jsx

```diff
@@ -1,40 +1,67 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 
 import MiqFormRenderer from '@@ddf';
 import createSchema from './physical-storage-form.schema';
 import miqRedirectBack from '../../helpers/miq-redirect-back';
 
-const PhysicalStorageForm = ({ redirect }) => {
-  const state = useState(undefined);
+const PhysicalStorageForm = ({ recordId, storageManagerId }) => {
+  const [state, setState] = useState({});
+  const {isLoading, initialValues} = state;
+  const submitLabel = !!recordId ? __('Save') : __('Add');
 
-  const onSubmit = async(values) => {
+  const loadSchema = (appendState = {}) => () => {
+    setState((state) => ({
+      ...state,
+      ...appendState,
+    }));
+  };
+
+  useEffect(() => {
+    if (recordId) {
+      API.get(`/api/physical_storages/${recordId}`).then((initialValues) => {
+        API.options(`/api/physical_storages?ems_id=${initialValues.ems_id}`).then(loadSchema({ initialValues: {...initialValues, edit: 'yes'}, isLoading: false }));
+      });
+    }
+  }, [recordId,]);
+
+  const onSubmit = ({ edit: _edit, ...values }) => {
     miqSparkleOn();
-    const message = __('Physical storage added');
-    API.post('/api/physical_storages', { action: 'create', resource: values })
-      .then(() => miqRedirectBack(message, 'success', redirect)).catch(miqSparkleOff);
+
+    const request = recordId ? API.patch(`/api/physical_storages/${recordId}`, values) : API.post('/api/physical_storages', values);
+    request.then(() => {
+      const message = sprintf(
+        recordId
+          ? __('Modification of Physical Storage "%s" has been successfully queued.')
+          : __('Add of Physical Storage "%s" has been successfully queued.'),
+        values.name,
+      );
+      miqRedirectBack(message, undefined, '/physical_storage/show_list');
+    }).catch(miqSparkleOff);
   };
 
   const onCancel = () => {
-    const message = __('Adding of physical storage was cancelled by the user');
-    miqRedirectBack(message, 'success', redirect);
+    const message = sprintf(
+      recordId
+        ? __('Edit of Physical Storage "%s" was canceled by the user.')
+        : __('Add of new Physical Storage was cancelled by the user.'),
+      initialValues && initialValues.name,
+    );
+    miqRedirectBack(message, 'warning', '/physical_storage/show_list');
   };
 
-  return (
+  return  (
     <MiqFormRenderer
-      schema={createSchema(...state)}
+      schema={createSchema(!!recordId, initialValues, state, setState)}
+      initialValues={initialValues}
+      canReset={!!recordId}
       onSubmit={onSubmit}
+      onReset={() => add_flash(__('All changes have been reset'), 'warn')}
       onCancel={onCancel}
-      buttonsLabels={{
-        submitLabel: __('Add'),
-        cancelLabel: __('Cancel'),
-      }}
+      buttonsLabels={{ submitLabel }}
     />
   );
 };
 
-PhysicalStorageForm.propTypes = {
```

Comment:
Please add propTypes and default props for this component. Example https://github.com/ManageIQ/manageiq-ui-classic/blob/eb8acc8762414b151b889ee77bbc753e8705c974/app/javascript/components/host-aggregate-form/index.jsx#L56

Reply:
done